lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -208,7 +220,21 @@ def add_included_resources_for(serializer, include_tree, primary_data, included)
           end
         end
 
-        def links_for(serializer, options)
+        def links_for(serializer)
+          serializer.links.each_with_object({}) do |(name, value), hash|
+            hash[name] =
+              if value.respond_to?(:call)
```

Comment:
then can you put a note in here about what sort of objects are expected, because if I have a Phone for my value, it's gonna error......